src/ol/interaction/modifyinteraction.js

```diff
@@ -976,31 +1067,36 @@ ol.interaction.Modify.prototype.insertVertex_ = function(segmentData, vertex) {
   }
 
   this.setGeometryCoordinates_(geometry, coordinates);
-  var rTree = this.rBush_;
```

Comment:
I can't figure out why it's OK to stop updating the R*-tree for all the other line and polygon type features, besides rectangle? Can you provide an explanation for this?